### ISSUE TITLE
Properties are not rendered by template when there is only one condition in allOf or anyOf

### ISSUE THREAD

**Issue by dejstheman** [coveooss/json-schema-for-humans#152]
https://github.com/coveooss/json-schema-for-humans/blob/main/json_schema_for_humans/templates/js/content.html#L45-L49

```html
        {%- if schema.kw_all_of and (schema.kw_all_of.array_items | length) == 1 -%}
            {{ content(schema.kw_all_of.array_items[0]) }}
        {%- elif schema.kw_any_of and (schema.kw_any_of.array_items | length) == 1 -%}
            {{ content(schema.kw_any_of.array_items[0]) }}
        {%- else -%}

            ......

            {%- for sub_property in schema.iterate_properties -%}
                {% include "section_properties.html" %}
            {%- endfor -%}
        {%- endif -%}
```

In `templates/js/content.html` we have these conditions to handle when there is only one condition in the schema but that also means because of the `else` it means that we dont render the properties at all.

**Comment by dblanchette** [comment]
I did not think that you could have `allOf`/`anyOf` and properties, but it seems perfectly legal.

I see three paths:
1. Remove the feature (`allOf` and `anyOf` will generate a single tab if it contains one element)
2. Merge when loading the schema (before it gets to the template)
e.g.
```json
{
  "allOf": [{"type": "string"}],
  "properties": {"a": {"type": "string"}}}
```
becomes
```json
{
  "type": "string",
  "properties": {"a": {"type": "string"}}
}
```
(well, the SchemaNode object for it)
3. Handle it in the template

I am tempted by option 1, it would be the caller responsibility to generate the schema the way they want it displayed
Option 2 is a bit too automagical (surprising behaviors) and could break something else I am not thinking of
Option 3 would not fix custom templates and requires porting that to all current and future "official" templates

Thoughts?

**Comment by dblanchette** [comment]
I went with option 2, released in version 0.42.1. Thanks for the report!
